kraken-common/src/kraken/common/supplier.py

```diff
@@ -136,8 +136,6 @@ def __repr__(self) -> str:
         return f"{self._value}.map({self._func})"
 
     def __eq__(self, other: object) -> bool:
-        if type(other) != type(self):
-            return False
```

Comment:
That's not equivalent to the current behaviour because 1. this raises an `AssertionError` instead of returning `False`, and 2. `isinstance` is looser than strict type checking (because it matches inherited objects as well).

I think the proper way to fix the flake8 error is to replace the type check with 

```python
        if type(self) is not type(other):
            return False
```

(using `is not` instead of `!=`)

Reply:
Indeed. my bad. Done.